cmd/sales-api/handlers/routes.go

```diff
@@ -10,7 +10,7 @@ import (
 )
 
 // API returns a handler for a set of routes.
-func API(log *log.Logger, masterDB *db.DB) http.Handler {
+func API(log *log.Logger, masterDB *db.DB, userAuth UserAuth) http.Handler {
```

Comment:
I am not sure value semantics make sense here? Do we want every function to have it's own copy?

Reply:
Yes because it's config. Maybe it could have a better name. `AuthConfig`?

Reply:
Would it make more sense to pass in a func or such `func GetUserToken(ctx context.Context, email, pass string) (user.Token, error)` down instead of the underlying private key? One benefit would be that the private key would not be passed down to all handlers, just the ability to make a token.

Reply:
I wouldn't want to pass a function in here. There are three options I would consider:

1. Pass in the config and let each layer pass down what is needed. This is the most explicit and is what we're doing now. This is also what is required by the current implementation of the auth platform package.
2. Redefine the auth package to make token generation a method on a type and have our call stack pass that type all the way down to our business layer. I floated that idea by Bill during a code review and we discussed that since the auth package doesn't need it to be a type+method we would just leave `auth.GenerateToken` as a function.
3. Do the same as option 2 but make this function depend on an interface. I don't see how that would really make the code any better though so I would have to see it implemented to know if I like that path.


In any event I want to delay making that decision right now. My next PR uses this same UserAuth type to wire up the authentication+authorization middlewares. Let's rehash this conversation on that PR when it's up because I think there is some improvement to be had.